### PR TITLE
refactor: simplify helm chart

### PR DESCRIPTION
**Pull request by the-wondersmith**
## 📝 Description

PR simplifies the repo's [helm chart](https://github.com/booklore-app/booklore/tree/5e13377966b4492b63f07c7ef4dfdee2cc2e9ecd/example-chart), bringing it much more in line with the existing [`docker-compose.yaml`](https://github.com/booklore-app/booklore/blob/develop/example-docker/docker-compose.yml) manifest.

**Linked Issue:** Addresses #2113 

## 🏷️ Type of Change

- [ ] Bug fix
- [ ] New feature
- [ ] Enhancement to existing feature
- [X] Refactor (no behavior change)
- [ ] Breaking change (existing functionality affected)
- [ ] Documentation update

## 🔧 Changes

- Remove MariaDB subchart
  - chart now directly uses the same `mariadb` image as the compose manifest [ref](https://github.com/booklore-app/booklore/blob/develop/example-docker/docker-compose.yml#L37)
- Templates now pass environment variables to the deployed `booklore` instance via standard Kubernetes `ConfigMap` and `Secret` machinery
  - additional user-specified values are passed through as well

## 🧪 Testing (MANDATORY)

> **PRs without this section filled out will be closed.** "Tests pass" or "Tested locally" is not sufficient. You must provide specifics.

**Manual testing steps you performed:**

1. `helm upgrade -i booklore example-chart --namespace booklore --create-namespace --set mariadb.config.password="$(tr -dc 'a-zA-Z0-9' < /dev/urandom | head -c 24)" --set mariadb.config.root_password="$(tr -dc 'a-zA-Z0-9' < /dev/urandom | head -c 24)"`
2.
3.

**Regression testing:**

-

**Edge cases covered:**

-

**Test output:**

<details>
<summary>Backend test output (<code>./gradlew test</code>)</summary>

```
PASTE OUTPUT HERE
```

</details>

<details>
<summary>Frontend test output (<code>ng test</code>)</summary>

```
PASTE OUTPUT HERE
```

</details>

## 📸 Screen Recording / Screenshots (MANDATORY)

> Every PR must include a **screen recording or screenshots** showing the change working end-to-end in a running local instance (both backend and frontend). This means you must have actually built, run, and tested the code yourself. PRs without visual proof will be closed without review.

---

## ✅ Pre-Submission Checklist

> **All boxes must be checked before requesting review.** Incomplete PRs will be closed without review. No exceptions.

- [ ] This PR is linked to an approved issue
- [ ] Code follows project [backend and frontend conventions](../CONTRIBUTING.md#backend-conventions)
- [ ] Branch is up to date with `develop` (merge conflicts resolved)
- [ ] I ran the full stack locally (backend + frontend + database) and verified the change works
- [ ] Automated tests added or updated to cover changes (backend **and** frontend)
- [ ] All tests pass locally and output is pasted above
- [ ] Screen recording or screenshots are attached above proving the change works
- [ ] PR is a single focused change (one bug fix OR one feature, not multiple unrelated changes)
- [ ] PR is reasonably scoped (PRs over 1000+ changed lines will be closed, split into smaller PRs)
- [ ] No unsolicited refactors, cleanups, or "improvements" are bundled in
- [ ] Flyway migration versioning is correct _(if schema was modified)_
- [ ] Documentation PR submitted to [booklore-docs](https://github.com/booklore-app/booklore-docs) _(if user-facing changes)_

### 🤖 AI-Assisted Contributions

> **If any part of this PR was generated or assisted by AI tools (Copilot, Claude, ChatGPT, etc.), all items below are mandatory.** You are fully responsible for every line you submit. "The AI wrote it" is not an excuse, and AI-generated PRs that clearly haven't been reviewed are the #1 reason PRs get closed.

- [ ] I have read and understand every line of this PR and can explain any part of it during review
- [ ] I personally ran the code and verified it works (not just trusted the AI's output)
- [ ] PR is scoped to a single logical change, not a dump of everything the AI suggested
- [ ] Tests validate actual behavior, not just coverage (AI-generated tests often assert nothing meaningful)
- [ ] No dead code, placeholder comments, `TODO`s, or unused scaffolding left behind by AI
- [ ] I did not submit refactors, style changes, or "improvements" the AI suggested beyond the scope of the issue

---

### CODE REVIEW

File: example-chart/templates/booklore/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    app.kubernetes.io/component: application
+    {{- include "booklore.labels" . | nindent 4 }}
+spec:
+  replicas: 1
+  strategy:
+    type: Recreate
+  selector:
+    matchLabels:
+      app.kubernetes.io/component: application
+      {{- include "booklore.labels" . | nindent 6 }}
```
Comment (acx10) [inline example-chart/templates/booklore/deployment.yaml:17]: This pulls in `helm.sh/chart` and `app.kubernetes.io/version` into `matchLabels`, which is immutable after creation. Next chart version bump would fail the upgrade. Should be `selectorLabels` here.

File: example-chart/templates/mariadb/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+  labels:
+    app.kubernetes.io/component: database
+    {{- include "booklore.labels" . | nindent 4 }}
+spec:
+  replicas: 1
+  strategy:
+    type: Recreate
+  selector:
+    matchLabels:
+      app.kubernetes.io/component: database
```
Comment (acx10) [inline example-chart/templates/mariadb/deployment.yaml:16]: Same thing here, `matchLabels` should use `selectorLabels` instead of the full label set.

File: example-chart/templates/mariadb/service.yaml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  name: mariadb
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app.kubernetes.io/component: database-service
+    {{- include "booklore.labels" . | nindent 4 }}
+spec:
+  type: {{ .Values.mariadb.service.type }}
+  selector:
+    app.kubernetes.io/component: database
+    {{- include "booklore.labels" . | nindent 4 }}
```
Comment (acx10) [inline example-chart/templates/mariadb/service.yaml:14]: Same selector issue as the deployments. Service selectors with version labels would stop matching pods after a chart bump. Should be `selectorLabels` here too.

File: example-chart/templates/config.yaml [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+type: Opaque
+metadata:
+  name: booklore-secrets
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app.kubernetes.io/component: secrets
+    {{- include "booklore.labels" . | nindent 4 }}
+  annotations:
+    helm.sh/hook: pre-install
+    helm.sh/hook-delete-policy: before-hook-creation
```
Comment (acx10) [inline example-chart/templates/config.yaml:13]: This being a `pre-install` hook means it won't get touched on `helm upgrade`. So if someone rotates a password in values and upgrades, the old secret stays. Might want `pre-upgrade` too, or just make it a regular managed resource?

File: example-chart/templates/_helpers.tpl [hunk 5]
```diff
@@ -47,7 +47,7 @@ Selector labels
 */}}
 {{- define "booklore.selectorLabels" -}}
 app.kubernetes.io/name: {{ include "booklore.name" . }}
-app.kubernetes.io/instance: {{ .Release.Name }}
+app.kubernetes.io/part-of: {{ .Release.Name }}
```
Comment (acx10) [inline example-chart/templates/_helpers.tpl:50]: `app.kubernetes.io/instance` is the conventional label for identifying a Helm release. Swapping it for `part-of` would break filtering in tools like Lens, k9s, Argo, etc. `part-of` is meant for grouping components of a larger app, not release identity. Any reason for the change?

File: example-chart/templates/booklore/pvc.yaml [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app.kubernetes.io/component: application-pvc
+    {{- include "booklore.labels" . | nindent 4 }}
+spec:
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 10Gi
```
Comment (acx10) [inline example-chart/templates/booklore/pvc.yaml:15]: The issue specifically called out StorageClass support. Both PVCs are missing `storageClassName`. An optional value would go a long way for anyone not using the default class.

**Review by acx10 — CHANGES_REQUESTED** [review]
The PR template sections (testing steps, regression testing, edge cases, test output, screenshots, and the pre-submission checklist) are mostly unfilled. Please complete those before this can move forward.

Also, the issue asked for StorageClass support on PVCs and the ability to pass extra env vars from external configMaps/secrets. The StorageClass piece is called out below, and the envFrom support is still missing.

**Comment by the-wondersmith** [comment]
> The PR template sections (testing steps, regression testing, edge cases, test output, screenshots, and the pre-submission checklist) are mostly unfilled. Please complete those before this can move forward.

Ah, yes, I know. It's why I haven't marked the PR ready for review yet 😅

> Also, the issue asked for StorageClass support on PVCs and the ability to pass extra env vars from external configMaps/secrets. The StorageClass piece is called out below, and the envFrom support is still missing.

I'll get 'em added shortly 🫡
